test/new-e2e/system-probe/test-runner/main.go

```diff
@@ -30,6 +31,34 @@ import (
 
 const matchAllPackages = "*"
 
+// testsuiteError wraps an exec.ExitError from a testsuite to preserve its exit code
+type testsuiteError struct {
+	exitErr  *exec.ExitError
+	exitCode int // custom exit code extracted from test output
+}
+
+func (e *testsuiteError) Error() string {
+	return e.exitErr.Error()
+}
+
+func (e *testsuiteError) Unwrap() error {
+	return e.exitErr
+}
+
+// extractExitCodeFromOutput parses output to find "###TEST_EXIT_CODE:X###" marker
```

Comment:
I think we could find a way to share the exit code of the testsuite with the test-runner without having to read its whole output.
Could we have the test-runner create a file, pass the path of this file to the testsuite as an argument, and then have the testsuite write its exit code to the file. This way the test-runner will just have to read the file to get the exit code? This also plays well with retries as the test-runner will read the exit code of the last testsuite run.